packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -249,19 +256,32 @@ function s(
                             ArrayPush.call(newChildren, vnode.factory(data.slotData, data.key));
                         });
                     } finally {
-                        setVMBeingRendered(vmBeingRenderedInception);
+                        setVMBeingRendered(vmBeingRendered);
                     }
                 } else {
+                    // This block is for standard slots (non-scoped slots)
+                    let clonedVNode;
+                    if (
+                        renderMode === RenderMode.Light &&
+                        (isVBaseElement(vnode) || isVStatic(vnode)) &&
+                        vnode.slotAssignment !== data.slotAssignment
```

Comment:
We only need to copy the vnodes when the slot assignment changes, copying every time causes issues with disconnected/connected callback firing.

I think it's related to some assumptions in the diffing algo during patching.

Reply:
Adding this as a comment would be helpful!

Reply:
Add in b68b82e6de9fd14f2c2f560b4f676f7b4157f536!